### PR TITLE
Rework session "Securing Communication"

### PR DESCRIPTION
**Pull request by razvand**
Add and rework content for "Securing Communication" section:
- [ ] Create session structure (add sections / subsections)
- [ ] Add links to `Further Reading` section
- [ ] Add challenge descriptions
- [ ] Add `Tools` section with tools used throughout the session

### CODE REVIEW

**Review by antonio-macovei — COMMENTED** [review]
I would add more text, making it sound like a story rather than a bullet point list. It's a little hard to follow in my opinion and doesn't match the rest of the sessions structure.
Maybe adding more context to links and commands would help.

**Comment by gabrielmocanu** [comment]
@razvand @antonio-macovei 

When do you think this can be merged?

**Comment by gabrielmocanu** [comment]
@razvand Don't forget to rename the session without index, something like 'securing-communication'.
Also rename`README.md` file to `index.md` and add this header[1] in the file in order for Docsy to be able to generate the right html file.

[1]: https://github.com/security-summer-school/web/blob/master/web-basics-browser-security-model/index.md#L1-L5

File: 03-securing-communication/README.md [hunk 1]
```diff
@@ -0,0 +1,480 @@
...
+```
+
+there would be no plain text output, because the connection is using a confidential (encrypted) channel.
+
+The same traffic inspection can be done in a more visual manner using Wireshark.
+
+As long as traffic is not encrypted, an attacker able to capture packets (either fooling someone to get the data or simply accessing a networking equipment along the way) will read the traffic contents.
+HTTPS uses public key cryptography to ensure the confidentiality of network traffic.
+
+## Public Key Cryptography. Certificates
```
Comment (gabrielmocanu) [inline 03-securing-communication/README.md:89]: I think we should add something about how private/public keys works on certificate signing because it might be confusing for that situation to see that encryption is made with private key.

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]

File: securing-communication/index.md [hunk 2]
```diff
@@ -0,0 +1,514 @@
...
+```
+
+there would be no plain text output, because the connection is using a confidential (encrypted) channel.
+
+The same traffic inspection can be done in a more visual manner using Wireshark.
+
+As long as traffic is not encrypted, an attacker able to capture packets (either fooling someone to get the data or simply accessing a networking equipment along the way) will read the traffic contents.
+HTTPS uses public key cryptography to ensure the confidentiality of network traffic.
+
+## Public Key Cryptography. Certificates
```
Comment (gabrielmocanu) [inline securing-communication/index.md:95]: I think you can delete certificates from this header and leave it in the next one.

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]

File: securing-communication/index.md [hunk 3]
```diff
@@ -0,0 +1,912 @@
...
+As this is is a self signed certificate, we use itself as CA.
+
+## HTTPS, SSL and TLS
+
+HTTPS is the secure version of HTTP providing identity management, confidentiality and integrity.
+HTTPS relies on at least the HTTP server providing a certificate that validates the identity.
+The client can also do that, as part of [client certificate authentication](https://comodosslstore.com/blog/what-is-ssl-tls-client-authentication-how-does-it-work.html).
+
+After validating the identity of the server, the client-server pair create a secure channel by agreeing on a per-session shared symmetric encryption key.
+This is negociated via a key-exchange algorithm such as Diffie-Hellman.
```
Comment (gabrielmocanu) [inline securing-communication/index.md:311]:
```suggestion
This is negotiated via a key-exchange algorithm such as Diffie-Hellman.
```

File: securing-communication/index.md [hunk 4]
```diff
@@ -0,0 +1,912 @@
...
+Then, all traffic between client and server will be encrypted.
+
+This entire process is facilited by the use of SSL (*Secure Sockets Layer*) or TLS (*Transport Layer Security*).
```
Comment (gabrielmocanu) [inline securing-communication/index.md:314]:
```suggestion
This entire process is facilitated by the use of SSL (*Secure Sockets Layer*) or TLS (*Transport Layer Security*).
```

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]
Don't forget to add `slides` directory and move `ca` and `https-capture-decrypt` to `assets` directory if you plan to keep those as reference in this session.

File: securing-communication/index.md [hunk 5]
```diff
@@ -0,0 +1,877 @@
...
+In HTTPS that means that if we connect to `google.com` we are offered guarantees that the server we connect to is indeed `google.com`.
+Otherwise, another server could impersonate the target server and capture all traffic.
+
+Identity management relies on signing and verifying messages using public-private keys.
+The private key is used to sign a message.
+The signed message is provided publicly.
+Then, the public key is used to verify the message.
+This is called a **digital signature**, as shown below:
+
+![digital signature](assets/digital-signature.svg)
```
Comment (gabrielmocanu) [inline securing-communication/index.md:146]: This image needs to be translated into English.

File: securing-communication/index.md [hunk 6]
```diff
@@ -0,0 +1,877 @@
...
+
+In HTTPS, this means that the web server will sign the message with its private key and web clients will verify the message with the public key.
+
+In order for this to work, the public key has to be attached the identity: the name of the server.
+This is done via a **digital certificate**.
+A certificate is a file that consists of a public key and an identity.
+A certificate itself is also signed to ensure its validity.
+This means that a certificate will also be verified using a public key, found as part of another certificate, as shown below:
+
+![digital certificate](assets/digital-certificate.png)
```
Comment (gabrielmocanu) [inline securing-communication/index.md:156]: This image needs to be translated into English.

File: securing-communication/index.md [hunk 7]
```diff
@@ -0,0 +1,877 @@
...
+$ openssl s_client -showcerts -connect www.google.com:443 -servername www.google.com < /dev/null 2> /dev/null | sed -ne '/-BEGIN CERTIFICATE-/,/-END CERTIFICATE-/p'
+```
+
+Assess remote HTTPS and certificate security:
+
+```
+$ ./testssl.sh security.cs.pub.ro
+```
+
+## Challenges
```
Comment (gabrielmocanu) [inline securing-communication/index.md:718]:
```suggestion
## Activities
```

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]

File: securing-communication/index.md (hunk 5 above)
Comment (razvand) [reply securing-communication/index.md:146]: Yes, I know, will keep it like this for now.

**Review by gabrielmocanu — APPROVED** [review]
